File: solutions/0031_Next_Permutation/iter.java
class Solution {
    public void nextPermutation(int[] nums) {
        if (nums.length <= 1) {
            return;
        }
        int i = nums.length - 2;
        while (i >= 0 && nums[i] >= nums[i + 1]) --i;
        if (i < 0) {
            Arrays.sort(nums);
            return;
        }
        int j = nums.length - 1;
        while (j > i && nums[j] <= nums[i]) --j;
        nums[i] = nums[i] ^ nums[j];
        nums[j] = nums[i] ^ nums[j];
        nums[i] = nums[i] ^ nums[j];
        Arrays.sort(nums, i + 1, nums.length);
    }
}

File: solutions/0485_Max_Consecutive_Ones/iter.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        best, cur = 0, 0
        for v in nums:
            if v == 0:
                best, cur = max(best, cur), 0
            else:
                cur += 1
        best = max(best, cur)
        return best
